internal/receiver/smartagentreceiver/log.go

```diff
@@ -242,3 +225,20 @@ func (l *logrusToZap) Fire(e *logrus.Entry) error {
 
 	return nil
 }
+
+func (l *logrusToZap) loggerMapValue0(src logrusKey) (*zap.Logger, bool) {
```

Comment:
is this function relocation worth the change?  seems like just bumping the logrus dep and adding the test case are the helpful bits.